ludwig/utils/data_utils.py

```diff
@@ -327,13 +331,15 @@ def save_hdf5(data_fp, data):
             h5_file.create_dataset(column, data=numpy_dataset[column])
 
 
-def load_hdf5(data_fp):
+def load_hdf5(data_fp, clean_cols: bool = False):
     with download_h5(data_fp) as hdf5_data:
         columns = [s.decode("utf-8") for s in hdf5_data[HDF5_COLUMNS_KEY][()].tolist()]
```

Comment:
For clarification, what happens if the columns key resolves to more hdf5 metadata?

Reply:
We assume `hdf5['columns']` will be flat and only contain the column names - I'm not aware of any cases where it contains extra metadata, but let me know if this is wrong

Reply:
I don't know first hand, but from reading it sounds like there could be arbitrary depth to group -> dataset expansion when reading hdf5 data. Let's close this PR and revisit if necessary.